### PR TITLE
RST-1277: Add S3 and create docker-support-services

### PR DESCRIPTION
**Pull request by Stivaros**
This PR adds minio as a private S3 server for ET3 and splits the various supporting Docker containers into `docker-support-services`, following the API pattern.

By adding minio we can add bring upload testing into ET3 itself as opposed to only doing it via full-system. This is based upon ticket RST-960 which had to be done in full-system by QA.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,8 +23,9 @@ Running this app will enable development of the app itself, with its test suite
 1) Install Docker and Docker Compose (links above)
 1) Change directory into the repo and run `bundle install` to install the gems
 1) Copy `.env.example` and rename to `.env`
-1) Run `docker-compose build`
+1) Run `./bin/dev/docker-support-services build`
```
Comment (garytaylor) [inline README.md]: Should this not be up not build ?

File: docker/dev/docker-compose.yml [hunk 2]
```diff
@@ -3,17 +3,27 @@ services:
...
     image: "dosel/zalenium"
-    container_name: zalenium
     hostname: zalenium
     tty: true
     volumes:
       - /tmp/videos:/home/seluser/videos
       - /var/run/docker.sock:/var/run/docker.sock
     ports:
-      - 4444:4444
+      - ${ZALENIUM_PORT:-4444}:4444
```
Comment (garytaylor) [inline docker/dev/docker-compose.yml]: Can this be SELENIUM_PORT ?   Its an emerging standard !!  We are not interested in what software is providing the selenium service

File: spec/support/setup_buckets.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+require 'aws-sdk-s3'
+
+RSpec.configure do |c|
+  c.before(:suite) do
+    minio_config = {
```
Comment (garytaylor) [inline spec/support/setup_buckets.rb]: Why minio_config ?  Its just S3 ?  we might change from minio one day and your code shouldnt really care.

Picky I know - just think our test suite shouldnt really care what S3 server it is using

File: README.md (hunk 1 above)
Comment (Stivaros) [reply README.md]: Thanks, fixed in 7f1eee8

File: docker/dev/docker-compose.yml (hunk 2 above)
Comment (Stivaros) [reply docker/dev/docker-compose.yml]: Fixed in 33ca404

File: spec/support/setup_buckets.rb (hunk 3 above)
Comment (Stivaros) [reply spec/support/setup_buckets.rb]: Good idea - fixed in 8c78e49

**Pull request merged**